lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1644,18 +1735,24 @@ LogicalResult ReduceOpConversion::matchAndRewriteBasic(
     ConversionPatternRewriter &rewriter) const {
   Location loc = op->getLoc();
   unsigned axis = op.axis();
+  bool withIndex = triton::ReduceOp::withIndex(op.redOp());
 
   auto srcTy = op.operand().getType().cast<RankedTensorType>();
   auto srcLayout = srcTy.getEncoding().cast<BlockedEncodingAttr>();
   auto srcOrd = srcLayout.getOrder();
   auto srcShape = srcTy.getShape();
 
   auto llvmElemTy = getTypeConverter()->convertType(srcTy.getElementType());
+  auto llvmIndexTy = rewriter.getI32Type();
```

Comment:
llvmIndexType should be got in a uniformed way from "getTypeConverter()->getIndexType()" instead of directly creating with I32Type.
In other contexts index type could also be lowered to I64 or other integer types.

Reply:
Thanks for pointing out this mistake. I will update the two `llvmIndexTy`s in `matchAndRewriteBasic` and `matchAndRewriteFast`

Reply:
Done